website/payments/api/v2/views.py

```diff
@@ -0,0 +1,145 @@
+from payments.api.v2 import filters
+from payments.api.v2.serializers import PaymentSerializer
+from payments.api.v2.serializers.payable_detail import PayableSerializer
+from payments.exceptions import PaymentError
+from payments.models import Payment, PaymentUser
+from thaliawebsite.api.v2.permissions import IsAuthenticatedOrTokenHasScopeForMethod
+from thaliawebsite.api.v2.serializers import EmptySerializer
+
+
+class PaymentListCreateView(ListAPIView, CreateAPIView):
+    queryset = Payment.objects.all()
+    serializer_class = PaymentSerializer
+
+    permission_classes = [IsAuthenticatedOrTokenHasScopeForMethod]
+    required_scopes_per_method = {
+        "GET": ["payments:read"],
+        "POST": ["payments:write"],
+    }
+    filter_backends = (
+        framework_filters.OrderingFilter,
+        filters.CreatedAtFilter,
+        filters.PaymentTypeFilter,
+    )
+    ordering_fields = ("created_at",)
+
+    def get_queryset(self):
+        return (
+            super()
+            .get_queryset()
+            .filter(paid_by=PaymentUser.objects.get(pk=self.request.member.pk))
+        )
+
+
+class PaymentDetailView(RetrieveAPIView):
+    queryset = Payment.objects.all()
+    serializer_class = PaymentSerializer
+    permission_classes = [IsAuthenticatedOrTokenHasScopeForMethod]
+    required_scopes_per_method = {"GET": ["payments:read"]}
+
+    def get_queryset(self):
+        return (
+            super()
+            .get_queryset()
+            .filter(paid_by=PaymentUser.objects.get(pk=self.request.member.pk))
+        )
+
+
+class PayableDetailView(RetrieveAPIView):
+    permission_classes = [IsAuthenticatedOrTokenHasScopeForMethod]
+    required_scopes_per_method = {
+        "GET": ["payments:read"],
+        "PUT": ["payments:write"],
+        "PATCH": ["payments:write"],
+    }
+
+    def get_serializer_class(self, *args, **kwargs):
+        if self.request.method.lower() == "put":
+            return EmptySerializer
+        return PayableSerializer
+
+    def get(self, request, *args, **kwargs):
+        serializer = self.get_serializer(self.get_payable())
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def get_payable(self):
+        app_label = self.kwargs["app_label"]
+        model_name = self.kwargs["model_name"]
+        payable_pk = self.kwargs["payable_pk"]
+
+        try:
+            payable_model = apps.get_model(app_label=app_label, model_name=model_name)
+            payable = payables.get_payable(
+                get_object_or_404(payable_model, pk=payable_pk)
+            )
+        except (LookupError, NotRegistered) as e:
+            raise serializers.ValidationError(
+                {api_settings.NON_FIELD_ERRORS_KEY: [_("Payable model not found")]}
+            ) from e
+
+        if (
+            not payable.payment_payer
+            or not payable.tpay_allowed
+            or payable.payment_payer != PaymentUser.objects.get(pk=self.request.user.pk)
+        ):
```

Comment:
I think you should be able to list the details of a non-tpay payment that you made yourself

Reply:
And when is that exactly?

Reply:
If you would want to check your payment history or something. For me it just feels like `HTTP GET thalia.nu/api/v2/payments` should return your payments.

Reply:
oh you meant payments that you were the payer of, I thought you referred to all payments that a user would also be the processor of. I am not sure if this comment is on the right line then, but what you are proposing already works.